### PR TITLE
Attacking a tendril rallies its mobs against you.

### PR DESCRIPTION
**Pull request by warriorstar-orion**
## What Does This PR Do
This PR makes it so that when a tendril is attacked, its existing mobs will be aggroed to you. There is a delay of 30 seconds between rally attempts, to prevent two players juggling the aggro, and allow miners to back off if necessary.

It also removes some dead code, `/mob/living/simple_animal/hostile/proc/summon_backup`, which was probably meant to do something similar but doesn't have the correct or robust target/faction/distance checking that exist in the FindTarget/GiveTarget/MoveToTarget procs on hostile animals.

I'm sure this behavior exists in other games but I got the idea from State of Decay 2, where plague hearts will defend themselves by sending zombies at their attackers.

## Why It's Good For The Game
As dgl said in Discord, "no more tendril sniping while the mobs ignore you". This honestly isn't that severe of a change; it just requires you to clear the mobs first or you might get overwhelmed by watcher freezes or legion skulls.
## Images of changes

https://github.com/ParadiseSS13/Paradise/assets/59303604/d913eed2-0257-440b-a84f-be707057a0e7

https://github.com/ParadiseSS13/Paradise/assets/59303604/ef510869-dd5e-4372-9aed-3b791917769f

## Testing
Spawned in, shot some tendrils, died.

## Changelog
:cl:
add: Lavaland tendrils, when attacked, will rally their spawn to retaliate. Take out their monsters to avoid being overwhelmed!
/:cl:

### CODE REVIEW

File: code/game/objects/structures/lavaland/necropolis_tendril.dm [hunk 1]
```diff
@@ -42,6 +44,26 @@ GLOBAL_LIST_EMPTY(tendrils)
 	new /obj/structure/closet/crate/necropolis/tendril(loc)
 	return ..()
 
+/obj/structure/spawner/lavaland/proc/rally_towards(mob/living/target)
+	var/datum/component/spawner/spawn_component = GetComponent(/datum/component/spawner)
```
Comment (Contrabang) [inline code/game/objects/structures/lavaland/necropolis_tendril.dm:48]: According to how components work, we should be minimizing use of `GetComponent`. Could you instead put this on the component and a new signal, maybe something like `COMSIG_SPAWNER_SET_TARGET`, and send that signal on where you currently have `rally_towards`?

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/lavaland/necropolis_tendril.dm (hunk 1 above)
Comment (warriorstar-orion) [reply code/game/objects/structures/lavaland/necropolis_tendril.dm:48]: Will attempt this, for now it's held back by the fact that the targeting code for hostile mobs has a blocking call on `tgui_input_text` for emoting. There may also be another issue where megafauna, who are all hostile subtypes despite not being meant for this code, use `SLEEP_CHECK_DEATH`, another thing not permitted in a signal handler.

Comment (Contrabang) [reply code/game/objects/structures/lavaland/necropolis_tendril.dm:48]: Yea, you'll have to use some INVOKE_ASYNC calls

Comment (warriorstar-orion) [reply code/game/objects/structures/lavaland/necropolis_tendril.dm]: Done and tested, seems to work well, PTAL?

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 2]
```diff
@@ -349,6 +349,12 @@
 	walk(src, 0)
 	LoseAggro()
 
+/// Shortcut proc to allow initiating combat slightly faster than waiting for
+/// normal processing.
```
Comment (Contrabang) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm:353]: I think autodoc only takes from the most recent ///
```suggestion
/// Shortcut proc to allow initiating combat slightly faster than waiting for normal processing.
```

**Review by Contrabang — APPROVED** [review]
LGTM

File: code/__DEFINES/dcs/signals.dm [hunk 3]
```diff
@@ -1041,3 +1041,7 @@
 
 ///from of mob/MouseDrop(): (/atom/over, /mob/user)
 #define COMSIG_DO_MOB_STRIP "do_mob_strip"
+
+
```
Comment (DGamerL) [inline code/__DEFINES/dcs/signals.dm:1045]:
```suggestion
```

File: code/datums/components/spawner.dm [hunk 4]
```diff
@@ -49,3 +51,13 @@
 	L.nest = src
 	L.faction = src.faction
 	P.visible_message("<span class='danger'>[L] [spawn_text] [P].</span>")
+
+/datum/component/spawner/proc/rally_spawned_mobs(parent, mob/living/target)
+	SIGNAL_HANDLER // COMSIG_SPAWNER_SET_TARGET
+
+	if(COOLDOWN_FINISHED(src, last_rally) && length(spawned_mobs))
```
Comment (DGamerL) [inline code/datums/components/spawner.dm:58]: Can you make an early return for this?

Comment (warriorstar-orion) [reply code/datums/components/spawner.dm]: done, PTAL

File: code/datums/components/spawner.dm [hunk 5]
```diff
@@ -49,3 +51,15 @@
...
+/datum/component/spawner/proc/rally_spawned_mobs(parent, mob/living/target)
+	SIGNAL_HANDLER // COMSIG_SPAWNER_SET_TARGET
+
+	if(!(COOLDOWN_FINISHED(src, last_rally) && length(spawned_mobs)))
+		return
+
+	// start the cooldown first, because a rallied mob might fire on
+	// ourselves while this is happening, causing confusion
+	COOLDOWN_START(src, last_rally, 30 SECONDS)
+	for(var/mob/living/simple_animal/hostile/rallied in spawned_mobs)
```
Comment (DGamerL) [inline code/datums/components/spawner.dm:64]:
```suggestion
	for(var/mob/living/simple_animal/hostile/rallied as anything in spawned_mobs)
```
Do all tendrils only spawn hostile mobs?

**Review by DGamerL — APPROVED** [review]

**Review by lewcc — APPROVED** [review]

**Pull request merged**